modules/asset-registry/src/lib.rs

```diff
@@ -130,15 +130,14 @@ pub mod pallet {
 		pub fn register_native_asset(
 			origin: OriginFor<T>,
 			currency_id: CurrencyId,
-			metadata: Box<AssetMetadata<BalanceOf<T>>>,
+			metadata: Box<AssetMetadata<BalanceOf<T>, String<T>>>,
```

Comment:
@vayesy since we've updated AssetMetadata's fields from Vec to BoundedVec, now the structure size is known and I believe it's no longer required to allocate memory on the heap. Can we remove Box here?